heron/schedulers/src/java/com/twitter/heron/scheduler/aurora/AuroraContext.java

```diff
@@ -34,4 +37,12 @@ public static String getHeronAuroraPath(Config config) {
     return config.getStringValue(JOB_TEMPLATE,
         new File(Context.heronConf(config), "heron.aurora").getPath());
   }
+
+  public static int getJobMaxKillAttempts(Config config) {
+    return config.getIntegerValue(JOB_MAX_KILL_ATTEMPTS, 5);
```

Comment:
do we put them in yaml?

Reply:
If people want to override the value, they would put it in the yaml file. 
Currently, we just put `5` as the default value and use it.

Reply:
i would suggest put an example config in yaml, otherwise how can user know there is a config like this in the code ?

Reply:
true. better to have an example.

Reply:
Added. Thanks.

Reply:
This value is hard coded, probably not a good idea. Since you already have the yaml config, you can use this value - whatever this value is. Since most of the time this value is not changed, it automatically becomes the default value. 

Also the logic seems to be incorrect. If you are allowing for overrides, the overridden value should be used - but is that there somewhere?

Reply:
- This value is hard coded, probably not a good idea.
We need to handle backward compatibility; this is why we have this value in source code. Different places would use their own configs, not necessarily sync up with OSS Heron repo. This implementation can prevent bringing down running Heron service.

- Since you already have the yaml config, you can use this value - whatever this value is. Since most of the time this value is not changed, it automatically becomes the default value.
We need to handle backward compatibility. This implementation also provides the flexibility about what you mentioned.

- If you are allowing for overrides, the overridden value should be used - but is that there somewhere?
`config.getIntegerValue(JOB_MAX_KILL_ATTEMPTS, 5);` means we would try to loop up the value for the key `JOB_MAX_KILL_ATTEMPTS` in the config. If people supply value for this key heron will use it, otherwise, it will use the default value 5. This implementation allows overrides, and the sample aurora/scheduler.yaml does the overrides.